src/document_array_get_attributes.py

```diff
@@ -0,0 +1,153 @@
+import os
+import string
+import pytest
+
+from faker import Faker
+import numpy as np
+
+
+from jina import Document, DocumentArray
+from jina.types.arrays.memmap import DocumentArrayMemmap
+from .utils.benchmark import benchmark_time
+
+fake = Faker()
+NUM_REPETITIONS = 5
+NUM_DOCS = 1000
+CHARS = tuple(string.ascii_uppercase + string.digits)
+
+
+def _generate_random_text():
+    return ''.join(np.random.choice(CHARS, 256))
+
+
+def _generate_random_blob():
+    return np.random.random(512)
+
+
+def _generate_random_buffer():
+    return bytes(bytearray(os.urandom(512 * 4)))
+
+
+def empty_docs():
+    return [Document() for _ in range(NUM_DOCS)]
+
+
+def text_docs(num_docs):
+    return [Document(text=_generate_random_text()) for _ in range(num_docs)]
+
+
+def blob_docs(num_docs):
+    return [Document(blob=_generate_random_blob()) for _ in range(num_docs)]
+
+
+def buffer_docs(num_docs):
+    return [Document(buffer=_generate_random_buffer()) for _ in range(num_docs)]
+
+
+def embedding_docs(num_docs):
+    return [Document(embedding=_generate_random_blob()) for _ in range(num_docs)]
+
+
+@pytest.mark.parametrize('memmap', [False, True])
+@pytest.mark.parametrize(
+    'field',
+    ['blob', 'text', 'buffer', 'embedding'],
```

Comment:
I would also put `docs_gen_fn` as a parameter together with `field` here, so you can take out `_get_docs`

Reply:
Done!